src/sortedtable/sortedtable.tsx

```diff
@@ -99,12 +110,108 @@ interface SortedTableState {
+  // indicating sort order. If not provided, the column is not considered
+  // sortable.
+  sortKey?: any;
+  // className is a classname to apply to the td elements
+  className?: string;
+  titleAlign?: "left" | "right" | "center";
+  // uniq column identifier
+  name: string;
+}
+
+/**
+ * SortSetting is the structure that SortableTable uses to indicate its current
+ * sort preference to higher-level components. It contains a sortKey (taken from
+ * one of the currently displayed columns) and a boolean indicating that the
+ * sort should be ascending, rather than descending.
+ */
+export interface SortSetting {
+  sortKey: any;
+  ascending: boolean;
+  columnTitle?: string;
+}
+
+/**
+ * TableProps are the props that should be passed to SortableTable.
+ */
+interface TableProps {
+  // Number of rows in the table.
+  count: number;
+  // Array of SortableColumns to render.
+  columns: SortableColumn[];
+  // Current sortSetting that was used to sort incoming data.
+  sortSetting?: SortSetting;
+  // Callback that should be invoked when the user want to change the sort
+  // setting.
+  onChangeSortSetting?: { (ss: SortSetting): void };
+  // className to be applied to the table element.
+  className?: string;
+  // A function that returns the class to apply to a given row.
+  rowClass?: (rowIndex: number) => string;
+  // expandableConfig, if provided, makes each row in the table "expandable",
+  // i.e. each row has an expand/collapse arrow on its left, and renders
+  // a full-width area below it when expanded.
+  expandableConfig?: ExpandableConfig;
+  firstCellBordered?: boolean;
+  renderNoResult?: React.ReactNode;
+  loading?: boolean;
+  loadingLabel?: string;
+  // empty state for table
+  empty?: boolean;
+  emptyProps?: EmptyProps;
+}
+
+export interface ExpandableConfig {
+  // Called when the expand toggle is clicked. If this prop is not supplied,
+  // the table is not expandable and the expansion control is not shown.
+  onChangeExpansion: (rowIndex: number, expanded: boolean) => void;
+  // Given a row index, return whether that row is expanded.
+  rowIsExpanded: (rowIndex: number) => boolean;
+  // If a row is expanded, this function is called to get the content for the
+  // full-width expanded section.
+  expandedContent: (rowIndex: number) => React.ReactNode;
+}
+
+/**
+ * SortableTable is designed to display tabular data where the data set can be
+ * sorted by one or more columns.
+ *
+ * SortableTable is not responsible for sorting data; however, it does allow the
+ * user to indicate how data should be sorted by clicking on column headers.
+ * SortableTable can indicate this to higher-level components through the
+ * 'onChangeSortSetting' callback property.
+ */
+
 export class SortedTable<T> extends React.Component<
   SortedTableProps<T>,
   SortedTableState
 > {
   static defaultProps: Partial<SortedTableProps<any>> = {
     rowClass: (_obj: any) => "",
+    count: 0,
```

Comment:
`count` value is calculated inside of the component (LINE 369). Do we need `count` prop or it can be removed?